appengine/standard/memcache/guestbook/main.py

```diff
@@ -85,8 +85,11 @@ def get_greetings(self, guestbook_name):
         greetings = memcache.get('{}:greetings'.format(guestbook_name))
         if greetings is None:
             greetings = self.render_greetings(guestbook_name)
-            if not memcache.add('{}:greetings'.format(guestbook_name),
-                                greetings, 10):
+            try:
+                if not memcache.add('{}:greetings'.format(guestbook_name),
+                                    greetings, 10):
+                    logging.error('Memcache set failed.')
```

Comment:
since these are two separate cases, you want two separate error messages. Should be, `Memcache add failed, key already exists` and `Memcache add failed, data larger than 1MB`